plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/blobstore/AzureBlobContainer.java

```diff
@@ -72,14 +72,14 @@ public InputStream readBlob(String blobName) throws IOException {
         logger.trace("readBlob({})", blobName);
 
         if (!blobExists(blobName)) {
```

Comment:
The better strategy is to change the mocks to conform to how the real APIs would behave (i.e. throw a 404 if the blob doesn't exist), in which case, the blobExists check is extraneous.  I will take care of this in a separate commit as part of the PR